internal/appsec/waf_test.go

```diff
@@ -426,3 +426,65 @@ func TestBlocking(t *testing.T) {
 		})
 	}
 }
+
+// Test that API Security schemas get collected when API security is enabled
+func TestAPISecurity(t *testing.T) {
+	// Start and trace an HTTP server
+	mux := httptrace.NewServeMux()
+	mux.HandleFunc("/apisec", func(w http.ResponseWriter, r *http.Request) {
+		pAppsec.MonitorParsedHTTPBody(r.Context(), "plain body")
+		w.Write([]byte("Hello World!\n"))
+	})
+	srv := httptest.NewServer(mux)
+	defer srv.Close()
+
+	req, err := http.NewRequest("POST", srv.URL+"/apisec?vin=AAAAAAAAAAAAAAAAA", nil)
+	require.NoError(t, err)
+
+	t.Run("enabled", func(t *testing.T) {
+		t.Setenv("DD_EXPERIMENTAL_API_SECURITY_ENABLED", "true")
+		t.Setenv("DD_API_SECURITY_REQUEST_SAMPLE_RATE", "1.0")
+		appsec.Start()
+		defer appsec.Stop()
+		if !appsec.Enabled() {
+			t.Skip("AppSec needs to be enabled for this test")
+		}
```

Comment:
And cases where appsec is explicitely disabled, for which it makes no sense to run this test. This is for example the case for a whole bunch of flavours in https://github.com/DataDog/dd-trace-go/blob/main/.github/workflows/appsec.yml. Since the env is set upstream in that matter it makes sense to perform this check.

Reply:
My problem with this approach is that this test may never run... and you most likely wouldn't know (spoilers: this happened to me several times).

The workflow differentiating between `DD_APPSEC_ENABLED` settings (`true`, `false`, blank/unset) was relevant when we had a build tag controlling the presence of appsec features in the binary (needed to verify it compiles correctly in all combinations).

If there is any interest in verifying the library behaves as intended with all possible/valid `DD_APPSEC_ENABLED` configurations... that would be an integration/end-to-end test, and NOT a unit test (which this is).

Reply:
c2700d646a1064fb